src/textual/widgets/_selection_list.py

```diff
@@ -679,6 +699,16 @@ def add_options(
                 raise SelectionError(
                     "Only Selection or a prompt/value tuple is supported in SelectionList"
                 )
+
+        # Add the new items to the value mappings.
+        self._values = {
+            **self._values,
+            **{
+                option.value: index
+                for index, option in enumerate(cleaned_options, start=self.option_count)
+            },
+        }
```

Comment:
Isn't `.update` better in terms of speed while being the same in terms of readability?

Reply:
Off the top of my head I don't know the speed difference; it feels like you are right, do you have a reference?

Reply:
I don't have a reference, no, it's just intuition maybe.
I was just surprised you spelled this out 🤷

Reply:
I'd probably go with update. It would be faster, but doubt the difference would be measurable. It's more that `update` expresses intent better.